### ISSUE TITLE
Error while appending additional elements or extending an existing named tuple.

### ISSUE THREAD

**Issue by qyuga** [microsoft/TypeScript#42106]
# Bug Report

### 🔎 Search Terms
named tuple, extending named tuples

named tuple, extend named tuple

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.0-dev.20201224#code/KYDwDg9gTgLgBDAnmYcAiAmAChAlgO3gF44BtEALjnwFcBbAI2CgBo5ErbHmBdAbgBQoSLATJUmOCVIA6OZhwEY-OHAD0auADkIcZlGhDw0eEhToAzIsJSycmQryE2AL070mUFaoEa4AUXBgAGMYYAATOCodAKgDKF9NAEFQmgBDABsqABUaMAzUOmBuKABnODoaUvhMjLgACzSAN1R8NKLy6Dha6gh4RpbqduBSmQEBIA
)

### 💻 Code

```ts

export type D2Point = [x: number, y: number];
export type D2 = [...D2Point];  // No error
export type D3Point = [...D2Point, z: number];   



```

### 🙁 Actual behavior

Compiler shows error  "Tuple members must all have names or all not have names."

### 🙂 Expected behavior
// Expected : No Error

**Comment by MartinJohns** [comment]
Looks like a duplicate of #39941. Used search terms: `named tuple spread`

**Comment by qyuga** [comment]
@MartinJohns They might be related. One difference is that #39941 didn't produce `Tuple members must all have names or all not have names.(5084)`.

**Comment by szymski** [comment]
I hit the same problem. I found an ugly workaround which looks like this:
`
type Append<A, B> =
    A extends [...infer Params] ? [...Params, ...(B extends [...infer Params2] ? Params2 : [])] : never;
`
Labels are preserved. But this issue should be fixed.
